lib/cdo/azure_content_moderator.rb

```diff
@@ -7,32 +7,45 @@
 # https://westus.dev.cognitive.microsoft.com/docs/services/57cf753a3f9b070c105bd2c1/operations/57cf753a3f9b070868a1f66c
 #
 class AzureContentModerator
+  class AzureError < StandardError; end
+  class RequestFailed < AzureError; end
+  class UnsupportedContentType < AzureError; end
+
   def initialize(endpoint:, api_key:)
     @endpoint = endpoint
     @api_key = api_key
   end
 
   #
-  # Give a publicly-accessible image URL, requests rating information from
-  # Azure Content Moderation and return a rating category.
+  # Give some binary image data and a content type, requests rating information
+  # from Azure Content Moderation and returns a rating category.
   #
-  # @param [String] image_url - must be reachable from the Azure service.
+  # @param [IO] image_data - binary image data to be rated
+  # @param [String] content_type - one of image/bmp, image/gif, image/jpeg, image/png
   # @returns [:everyone|:racy|:adult]
-  def rate_image(image_url)
+  #
+  def rate_image(image_data, content_type)
+    raise UnsupportedContentType.new("Cannot accept content-type #{content_type}") unless %w(
+      image/bmp
+      image/gif
+      image/jpeg
+      image/png
+    ).include? content_type
```

Comment:
Apologies for not fully thinking through this on my own, but it looks like we support other content types in other places. for external urls:
 https://github.com/code-dot-org/code-dot-org/blob/ce7f271e46dc6a41b80782b96b2a832d5d2dc684/dashboard/app/controllers/media_proxy_controller.rb#L18-L31
is this going to be a problem?

Reply:
The [API docs for Azure](https://westus.dev.cognitive.microsoft.com/docs/services/57cf753a3f9b070c105bd2c1/operations/57cf753a3f9b070868a1f66c) explicitly call these out as the only supported content types.  Knowing that, it makes sense for us to catch this on our end before making the request.

For now it's a nonissue anyway - all our thumbnails are pngs.  Just trying to future-proof a bit.